File: scalaproject/src/main/scala/com/lab/chapter01/test/TestScala.java
package com.lab.chapter01.test;

/**
 * Created by 17081290 on 2020/5/14.
 */
//scala运行时进行包装
public class TestScala {
    public static void main(String[] paramArrayOfString) {
        System.out.println("TestScala method .....");
        //TestScala..MODULE$.main(paramArrayOfString);
        TestScala$.MODULE$.main(paramArrayOfString);
    }
}

final class TestScala$ {
    public static final TestScala$ MODULE$;
    static {
        MODULE$ = new TestScala$();
    }

    public void main(String[] args) {
        System.out.println("hello");
    }
//    private TestScala$() {
//        MODULE$ = this;
//    }
}

File: scalaproject/src/main/scala/com/lab/chapter01/TestScala.scala
package com.lab.chapter01


/**
  * 1.只要有object,则应该想到的是
  * A.object TestScala 对应的是一个TestScala$的对象MODULE$
  * B.在程序中,它是个单例
  */
object TestScala {
  def main(args: Array[String]) {
    println("hello")
  }
}
